src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -1383,5 +1377,123 @@ public C()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T> where T : class
+{
+    [MaybeNull] public T P { get; set; }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_ClassT_Field_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T> where T : class
+{
+    [MaybeNull] public T P;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Property_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P { get; set; }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Field_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Property_NullInitializer()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P { get; set; } = null;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (4,49): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     [MaybeNull] public string P { get; set; } = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 49)
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Field_NullInitializer()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P = null;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (4,35): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     [MaybeNull] public string P = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 35)
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

Minor point: I feel like it would be easier to read if the new tests were combined into one or two tests, with multiple properties and fields.